tests/cloudpickle_test.py

```diff
@@ -614,13 +614,33 @@ def test_NotImplementedType(self):
         res = pickle_depickle(type(NotImplemented), protocol=self.protocol)
         self.assertEqual(type(NotImplemented), res)
 
-    def test_builtin_function_without_module(self):
-        on = object.__new__
-        on_depickled = pickle_depickle(on, protocol=self.protocol)
-        self.assertEqual(type(on_depickled(object)), type(object()))
+    @pytest.mark.skipif(
+        sys.version_info[0] >= 3,
+        reason="builtin_function_or_method are special-cased only in python2")
+    def test_builtin_function(self):
+        # builtin function from the __builtin__ module
+        assert pickle_depickle(zip, protocol=self.protocol) is zip
+
+        from sys import getcheckinterval
+        # builtin function from a "regular" module
+        assert pickle_depickle(
+            getcheckinterval, protocol=self.protocol) is getcheckinterval
+
+    @pytest.mark.skipif(
+        sys.version_info[0] >= 3,
+        reason="builtin_function_or_method are special-cased only in python2")
+    def test_builtin_method(self):
+        # pickle_depickle some builtin methods of the __builtin__ module
+        for t in list, tuple, set, frozenset, dict, object:
+            cloned = pickle_depickle(t.__new__, protocol=self.protocol)
+            self.assertTrue(cloned is t.__new__)
 
+        # pickle_depickle a method of a "regular" module
         fi = itertools.chain.from_iterable
         fi_depickled = pickle_depickle(fi, protocol=self.protocol)
+
+        # fi is fi_depickled will return false, but so does
+        # itertools.chain.from_iterable is itertools.chain.from_iterable
```

Comment:
Maybe put backticks around the Python expression, otherwise it's hard to parse what is Python and what is English in this comment:

```python
        # `fi is fi_depickled` would return False, but so will
        # `itertools.chain.from_iterable is itertools.chain.from_iterable`.
        # Instead of testing physical equality, check that `fi_depickled` behaves as
        # expected.
```